### PR TITLE
Fix for is_primary= false ignore in VM resource

### PR DESCRIPTION
**Pull request by vbauzysvmware**
Ref: https://github.com/vmware/terraform-provider-vcd/issues/549

simple fix for strange behaviour - value `false` was treated the same as wasn't provided by user.

### CODE REVIEW

File: vcd/resource_vcd_vapp_vm_multinetwork_test.go [hunk 1]
```diff
@@ -170,7 +170,7 @@ func TestAccVcdVAppVmMultiNIC(t *testing.T) {
 
 					resource.TestCheckResourceAttr("vcd_vapp_vm."+netVmName1, "network.1.name", "multinic-net"),
 					resource.TestCheckResourceAttr("vcd_vapp_vm."+netVmName1, "network.1.type", "org"),
-					resource.TestCheckResourceAttr("vcd_vapp_vm."+netVmName1, "network.1.is_primary", "true"),
+					resource.TestCheckResourceAttr("vcd_vapp_vm."+netVmName1, "network.1.is_primary", "false"),
```
Comment (Didainius) [inline vcd/resource_vcd_vapp_vm_multinetwork_test.go:173]: Looking at definition of `testAccCheckVcdVAppVmNetworkVmStep1` adaper #1 has `is_primary         = true`. Is this valid?

Comment (vbauzysvmware) [reply vcd/resource_vcd_vapp_vm_multinetwork_test.go:173]: I made update in HCL, 0 Net get false, 1 Net gets true.

**Comment by Didainius** [comment]
At least the below tests failed for me in 9.7 and 10:
```
--- FAIL: TestAccVcdMediaInsertBasic (537.69s)
--- FAIL: TestAccVcdVappFirewallRules (1856.47s)
--- FAIL: TestAccVcdVAppMultiVmInTemplate (1472.44s)
--- FAIL: TestAccVcdVappNatRules (724.04s)
--- FAIL: TestAccVcdVAppRaw_Basic (682.78s)
--- FAIL: TestAccVcdVAppVmMulti (1704.54s)
--- FAIL: TestAccVcdVAppVm_Basic (593.48s)
--- FAIL: TestAccVcdVAppVm_Clone (896.23s)
--- FAIL: TestAccVcdVAppVmCreateCustomizationFalse (542.32s)
--- FAIL: TestAccVcdNsxvEdgeFirewallRuleVms (578.11s)
```

**Comment by vbauzysvmware** [comment]
> At least the below tests failed for me in 9.7 and 10.0:
> 
> ```
> --- FAIL: TestAccVcdMediaInsertBasic (537.69s)
> --- FAIL: TestAccVcdVappFirewallRules (1856.47s)
> --- FAIL: TestAccVcdVAppMultiVmInTemplate (1472.44s)
> --- FAIL: TestAccVcdVappNatRules (724.04s)
> --- FAIL: TestAccVcdVAppRaw_Basic (682.78s)
> --- FAIL: TestAccVcdVAppVmMulti (1704.54s)
> --- FAIL: TestAccVcdVAppVm_Basic (593.48s)
> --- FAIL: TestAccVcdVAppVm_Clone (896.23s)
> --- FAIL: TestAccVcdVAppVmCreateCustomizationFalse (542.32s)
> --- FAIL: TestAccVcdNsxvEdgeFirewallRuleVms (578.11s)
> ```

Fixed

**Comment by Didainius** [comment]
Tests pass, but manual test failed.

Consider the following example (VM with 2 NICs - 0,1):



```hcl
resource "vcd_vapp_vm" "vm1" {
...
  network {
    type               = "org"
    name               = vcd_vapp_org_network.vappNetwork1.org_network_name
    ip_allocation_mode = "MANUAL"
    ip                 = "10.10.102.161"
  }


  network {
    type               = "org"
    name               = vcd_vapp_org_network.vappNetwork1.org_network_name
    ip_allocation_mode = "MANUAL"
    ip                 = "10.10.102.162"
    is_primary         = true
  }
...
```


* (initial) apply. is_primary=true on NIC1 only (NIC0 does not have flag `is_primary` at all)
* 2 apply `is_primary=true` on NIC0 only. (NIC1 does not have flag)
#  Plan shows it and update occurs, but NIC1 remains primary (even after many applies). 
I think some issue still lies in `func networksToConfig`

**Comment by vbauzysvmware** [comment]
> ```terraform
> is_primary
> ```

this is expected - field is `computed` and `optional`, you removed -it stayed as true.

**Comment by Didainius** [comment]
> > ```terraform
> > is_primary
> > ```
> 
> this is expected - field is `computed` and `optional`, you removed -it stayed as true. If you want good behaviour in such case, explicitly define values.

I removed only from the second NIC (index 1). NIC one (index 0) had `is_primary=true` and "apply" showed diff, but it did not take effect.

**Comment by vbauzysvmware** [comment]
> > > ```terraform
> > > is_primary
> > > ```
> > 
> > 
> > this is expected - field is `computed` and `optional`, you removed -it stayed as true. If you want good behaviour in such case, explicitly define values.
> 
> I removed only from the second NIC (index 1). NIC one (index 0) had `is_primary=true` and "apply" showed diff, but it did not take effect.

Could you tell me step by step what I should reproduce?

**Comment by vbauzysvmware** [comment]
> > > > ```terraform
> > > > is_primary
> > > > ```
> > > 
> > > 
> > > this is expected - field is `computed` and `optional`, you removed -it stayed as true. If you want good behaviour in such case, explicitly define values.
> > 
> > 
> > I removed only from the second NIC (index 1). NIC one (index 0) had `is_primary=true` and "apply" showed diff, but it did not take effect.
> 
Made changes on which we agreed. Now works better, thank you

**Review by Didainius — APPROVED** [review]
Finally. Thanks!

**Review by dataclouder — APPROVED** [review]

**Pull request merged**